functions.cpp

```diff
@@ -408,15 +408,19 @@ namespace Sass {
       HSL hsl_color = rgb_to_hsl(rgb_color->r(),
                                  rgb_color->g(),
                                  rgb_color->b());
-      //Check saturation is not negative before saturate it
-      double hslcolorS = hsl_color.s;
+
+      double hslcolorS = hsl_color.s + amount->value();
```

Comment:
This is subtly different, but maybe that's intended. Originally for negative input `-X%` saturated by `Y%`: `-X%` -> `0%` -> `Y%`, but now `-X%` -> `(-X+Y)%` (which is likely not the same as `Y%`).